### PR TITLE
Add procedure how to store sensitive strings with credential stores

### PR DESCRIPTION
**Pull request by dvagnero**
Docs for #1730

### CODE REVIEW

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+.Prerequisites
+* Have valid {brandname} configuration in XML, YAML, or JSON format.
+
+.Procedure
+. Create a `CredentialStore Secret` file.
+. Specify the credentials and its aliases with the `data` field.
++
+.User-secret.yaml
+[source,yaml]
+----
```
Comment (dvagnero) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:17]: Need to move both examples in a separate file.

**Comment by dvagnero** [comment]
What is the right terminology? credential store? credential keystore?

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
+[id='secure_credentials_{context}']
+= Securing custom {brandname} configuration
+
+[role="_abstract"]
+Define and store custom {brandname} configuration that requires credentials securely.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:5]: I don't think the last part of the sentence makes sense. How about something like:

"Securely define and store custom {brandname} server configurations"?

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+To protect sensitive text strings such as passwords, add the entries in a credential store rather than directly in the `Infinispan` CR.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:6]: Credentials would be stored in the server configuration `xml|json|yaml`, not the Infinispan CR itself.

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+.Prerequisites
+* Have valid {brandname} configuration in XML, YAML, or JSON format.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:9]: "Have a valid.."?

**Review by ryanemerson — COMMENTED** [review]
Thanks @dvagnero, it reads well. Just a few minor comments

**Comment by dvagnero** [comment]
Thank you @ryanemerson 
Ready for your review

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+. Specify the credentials and its aliases with the `data` field.
++
+.User-secret.yaml
+[source,yaml]
+----
+include::yaml/credential_store_secret.yaml[]
+----
+
+. Open the `Infinispan` CR for editing.
+. Specify the name of the credential store in the `spec.security.credentialStoreSecretName` field.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:22]: "...name of hte credential store Secret in the..."

File: documentation/asciidoc/topics/proc_securing_custom_configuration.adoc [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+[role="_abstract"]
+Securely define and store custom {brandname} Server configuration.
+To protect sensitive text strings such as passwords, add the entries in a credential store rather than directly in the {brandname} Server configuration.
+
+.Prerequisites
+* Have a valid {brandname} configuration in XML, YAML, or JSON format.
+
+.Procedure
+. Create a `CredentialStore Secret` file.
+. Specify the credentials and its aliases with the `data` field.
```
Comment (ryanemerson) [inline documentation/asciidoc/topics/proc_securing_custom_configuration.adoc:13]: I think we need to say to apply the Secret file after this stage.

**Comment by ryanemerson** [comment]
Thanks @dvagnero

**Pull request merged**